File: Catalyst-Mono/Catalyst-DiscordCore/src/main/java/net/kingchev/catalyst/ru/command/model/CatalystCommand.java
package net.kingchev.catalyst.ru.command.model;

import net.dv8tion.jda.api.Permission;
import org.springframework.stereotype.Component;

import java.lang.annotation.*;

@Component
@Inherited
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
public @interface CatalystCommand {
    String key();

    String description();

    Permission[] botPermissions() default { Permission.MESSAGE_SEND, Permission.MESSAGE_EMBED_LINKS };

    Permission[] userPermissions();

    String group() default "discord.group.misc";

    int priority() default 1;

    boolean hidden() default false;

    boolean developerOnly() default false;

    boolean guildOnly() default true;

    boolean dmOnly() default false;
}

File: Catalyst-Mono/Catalyst-Core/src/main/kotlin/net/kingchev/catalyst/ru/discord/command/model/CatalystCommand.kt
package net.kingchev.catalyst.ru.discord.command.model

import net.dv8tion.jda.api.Permission
import org.springframework.stereotype.Component
import java.lang.annotation.Inherited

@Inherited
@Component
@Retention(AnnotationRetention.RUNTIME)
@Target(allowedTargets = [AnnotationTarget.CLASS])
annotation class CatalystCommand(
    val key: String,
    val aliases: Array<String> = [],
    val description: String,
    val permissions: Array<Permission> = [Permission.MESSAGE_SEND, Permission.MESSAGE_EMBED_LINKS],
    val userPermission: Array<Permission> = [],
    val group: String,
    val hidden: Boolean = false,
    val developerOnly: Boolean = false,
    val guildOnly: Boolean = true,
    val dmOnly: Boolean = false
)
